src/main.cpp

```diff
@@ -790,6 +814,43 @@ int main ( int argc, char** argv )
 //CTestbench Testbench ( "127.0.0.1", DEFAULT_PORT_NUMBER );
     // clang-format on
 
+    CRpcServer* pRpcServer = nullptr;
+
+    if ( iJsonRpcPortNumber != INVALID_PORT )
+    {
+        QString strJsonRpcSecret = [] ( QString strJsonRpcSecretFileName ) -> QString {
```

Comment:
Please stick to a consistent code style for the file being edited.

Reply:
@pljones Everything is formatted by Clang-Format — how would you change this code to make it more “consistent”?

Reply:
Clang format only checks spacing, not the approach.

What I mean is that main.cpp isn't written like this.  Either overhaul the whole file in a separate change or stick to the existing way of working.

Reply:
Do you mean the lambda usage? I think I introduced it here. The reason was that this block lead to *very* deep nesting otherwise. The lambda allows using a `return` statement to avoid that. So, the options I see are:
- Deeply nested `if`s, but in style with the rest of the file
- The lambda, which is not in style with the rest of the file but keeps the code most readable (IMO)
- There may be other solutions (artificial loops? I have to resist mentioning `goto`... ;)), which didn't look too nice to me.

Reply:
Marking this as resolved. The fact that we decided to make --jsonrpcsecretfile-related failures fatal made it possible to remove the lambdas without any of the other downsides. 7e5317ce332ee92ecd9787a0188c9c502d1c1bdb